testsuite/MDAnalysisTests/core/test_accumulate.py

```diff
@@ -0,0 +1,165 @@
+# -*- Mode: python; tab-width: 4; indent-tabs-mode:nil; coding:utf-8 -*-
+# vim: tabstop=4 expandtab shiftwidth=4 softtabstop=4 fileencoding=utf-8
+#
+# MDAnalysis --- https://www.mdanalysis.org
+# Copyright (c) 2006-2017 The MDAnalysis Development Team and contributors
+# (see the file AUTHORS for the full list of names)
+#
+# Released under the GNU Public Licence, v2 or any higher version
+#
+# Please cite your use of MDAnalysis in published work:
+#
+# R. J. Gowers, M. Linke, J. Barnoud, T. J. E. Reddy, M. N. Melo, S. L. Seyler,
+# D. L. Dotson, J. Domanski, S. Buchoux, I. M. Kenney, and O. Beckstein.
+# MDAnalysis: A Python package for the rapid analysis of molecular dynamics
+# simulations. In S. Benthall and S. Rostrup editors, Proceedings of the 15th
+# Python in Science Conference, pages 102-109, Austin, TX, 2016. SciPy.
+# doi: 10.25080/majora-629e541a-00e
+#
+# N. Michaud-Agrawal, E. J. Denning, T. B. Woolf, and O. Beckstein.
+# MDAnalysis: A Toolkit for the Analysis of Molecular Dynamics Simulations.
+# J. Comput. Chem. 32 (2011), 2319--2327, doi:10.1002/jcc.21787
+#
+from __future__ import division, absolute_import
+
+import numpy as np
+from numpy.testing import assert_equal, assert_almost_equal
+
+import MDAnalysis as mda
+from MDAnalysis.exceptions import DuplicateWarning, NoDataError
+from MDAnalysisTests.datafiles import PSF, DCD, TPR_xvf, TRR_xvf, GRO
+import pytest
+
+universe = mda.Universe(PSF, DCD)
+universe_molfrag = mda.Universe(TPR_xvf, TRR_xvf)
+levellist = ('atoms', 'residues', 'segments')
```

Comment:
I added the tests for segments; However, I am not happy with putting the decorator `@pytest.mark.parametrize('level', levellist)` in front of **every** function in the class but unfortunately I haven't found a better solution. Maybe somebody who knows pytest better finds a smarter way...

Reply:
You can place the decorator at the class level, but then **every** test in the class needs to use it

```python
@pytest.mark.parametrize('this', [1, 2])
class TestThing(object):
    def test_add(self, this):
        something
```

Or you can make a fixture have a loop in it using params

```python
@pytest.fixture(params=[1, 2])
def thing(request):
    return request.param[0]

def test_this(thing):
    assert thing == 1 or thing == 2
```

Reply:
Nice. Thanks your first suggestion will do the job here.